### PR TITLE
Implement  #174 to handle additional token errors.

### PR DESCRIPTION
**Pull request by HofmannZ**
As described in #174, there are a couple of scenarios where `verifyIdToken` was throwing an error, where it could actually be handled more gracefully.

This PR:

- implements the suggestions in #174,
- upgrades the `firebase` and `firebase-admin` dependencies, and
- upgrades other dependencies that have non-breaking changes.

This PR also brings RFC issue #265 one step closer to being ready for launch 🚀.

### CODE REVIEW

**Comment by HofmannZ** [comment]
@kmjennison - Ran the failing test locally without my changes, and it fails with the same error. Any idea why that might be?

**Comment by HofmannZ** [comment]
@kmjennison - Yeah they do.

**Comment by kmjennison** [comment]
@HofmannZ Hmm, ok. I don't have much time to debug at the moment but will try to take a look when I can.

**Comment by HofmannZ** [comment]
@kmjennison - Ready for review! 👀

File: src/firebaseAdmin.js [hunk 1]
```diff
@@ -59,13 +52,30 @@ export const verifyIdToken = async (token, refreshToken = null) => {
...
+        break
+      case 'auth/invalid-user-token':
+      case 'auth/user-token-expired':
+      case 'auth/user-disabled':
+        // Return an unauthenticated user
+        newToken = null
+        firebaseUser = null
+        break
+      default:
+        // Otherwise, throw.
```
[the remark this answers is not in the document]
Comment (HofmannZ) [reply src/firebaseAdmin.js:77]: Sounds like a good idea, bit out of scope for this PR. Will add a new PR for that when I find the time for it.

[the remark this answers is not in the document]
Comment (kmjennison) [reply src/firebaseAdmin.js:77]: Thanks, I plan to take this work on soon.

**Comment by kmjennison** [comment]
Thanks, @HofmannZ! Merged in #361.

File: src/firebaseAdmin.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kmjennison) [reply src/firebaseAdmin.js:77]: Implemented this in #366.

**Pull request closed without merging**